### PR TITLE
Integrate with Tower Workflow

### PR DESCRIPTION
**Pull request by jameswnl**
https://bugzilla.redhat.com/show_bug.cgi?id=1468339

Depends on https://github.com/ManageIQ/manageiq/pull/17440

Part of https://github.com/ManageIQ/manageiq/issues/17441

### CODE REVIEW

**Comment by jameswnl** [comment]
@Ladas can you take a look? thanks

File: app/models/manageiq/providers/ansible_tower/shared/inventory/parser/automation_manager.rb [hunk 1]
```diff
@@ -109,4 +111,29 @@ def credentials
...
+
+  def workflow_nodes
+    collector.workflow_job_template_nodes.each do |node|
+      inventory_object = persister.workflow_nodes.find_or_build(node.id.to_s)
+      inventory_object.workflow = persister.workflows.lazy_find(node.workflow_job_template_id.to_s)
+      inventory_object.configuration_script = persister.configuration_scripts.lazy_find(node.unified_job_template_id)
+      %w(success failure always).each do |condition|
+        node.send("#{condition}_nodes_id").each do |child_id|
+          child_node = persister.workflow_nodes.lazy_find(child_id)
+          # child_node.parent = inventory_object
```
Comment (jameswnl) [inline app/models/manageiq/providers/ansible_tower/shared/inventory/parser/automation_manager.rb]: @Ladas How do I get Line 133-134 to work? 
or @agrare ?

Comment (Ladas) [reply app/models/manageiq/providers/ansible_tower/shared/inventory/parser/automation_manager.rb]: hm, doing this from this direction, you could just

```ruby
child_node = persister.workflow_nodes.find_or_build(child_id)
child_node.parent = inventory_object
```

then the upper find_or_build should fill the rest of the info, when it reaches the node. How is the parent implemented? Is it just the foreign key?

Comment (jameswnl) [reply app/models/manageiq/providers/ansible_tower/shared/inventory/parser/automation_manager.rb]: Great! Thanks @Ladas

**Comment by jameswnl** [comment]
@miq-bot remove_label wip
@miq-bot add_labels enhancement

**Pull request merged**